### DISCUSSION TITLE
Why am I getting E_MISSING_ENV_VALUE when I have .env file in root when running migration?

### DISCUSSION THREAD

**Discussion by matthewswallace** [adonisjs/core#3692] [Help] [answered]
I have the following .env file setup in the root of my project. 

```
PORT=3333
HOST=127.0.0.1
NODE_ENV=development
APP_KEY=<Key>
DRIVE_DISK=local
SESSION_DRIVER=cookie
CACHE_VIEWS=false
DB_CONNECTION=mysql
MSSQL_SERVER=
MYSQL_HOST=localhost
MYSQL_PORT=3306
MYSQL_USER=root
MYSQL_PASSWORD=
MYSQL_DB_NAME=fulcrum
```
I'm using lucid and auth plugins and a MySQL db. 

My env.ts file looks like this.
```
import Env from '@ioc:Adonis/Core/Env'

export default Env.rules({
	HOST: Env.schema.string({ format: 'host' }),
	PORT: Env.schema.number(),
	APP_KEY: Env.schema.string(),
	APP_NAME: Env.schema.string(),
	CACHE_VIEWS: Env.schema.boolean(),
	SESSION_DRIVER: Env.schema.string(),
  DRIVE_DISK: Env.schema.enum(['local'] as const),
	NODE_ENV: Env.schema.enum(['development', 'production', 'test'] as const),
  MSSQL_SERVER: Env.schema.string({ format: 'host' }),
  MSSQL_PORT: Env.schema.number(),
  MSSQL_USER: Env.schema.string(),
  MSSQL_PASSWORD: Env.schema.string.optional(),
  MSSQL_DB_NAME: Env.schema.string(),

})
```
Here is what my project structure looks like 
![Screen Shot 2022-04-24 at 10 45 40 AM](https://user-images.githubusercontent.com/123816/164984649-31662e36-2973-4ed4-b7d8-ea1e7d943367.png)

**Comment by RomainLanz** [comment, marked as answer]
Hey @matthewswallace! 👋🏻 

The error is more than just `E_MISSING_ENV_VALUE`, it tells you that you are requiring some environment variables but you didn't defined them.

Looking at the info you shared, you are validating `MSSQL` variable, but you are using `MYSQL`.

**Reply by matthewswallace** [reply]
🤪  copy paste error on my part. Thanks!
